app/helpers/omniauth_helper.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+module OmniauthHelper
+  def check_provider(auth)
+    user = User.find_by(email: auth.info.email)
+    if Provider.find_by(uid: auth.uid, provider: auth.provider)
```

Comment:
uid liệu có thay đổi đc ko :-? nếu có thì giữ các cũ ko có tác dụng gì 
sao ko tìm theo user_id + provider 
về bản chất provider belong to user nên tìm theo user okie hơn